Frends.AzureBlobStorage.ListBlobsInContainer/Frends.AzureBlobStorage.ListBlobsInContainer/ListBlobsInContainer.cs

```diff
@@ -0,0 +1,151 @@
+        /// <returns>List of blobs in container with chosen listing structure.</returns>
+        /// <exception cref="Exception">Authentication exception.</exception>
+
+        public static async Task<Result> ListBlobsInContainer([PropertyTab] Source source)
+        {
+            BlobContainerClient blobContainerClient = null;
+            var blobsList = new List<Result>();
+            var authSas = false;
+
+            switch (source.AuthenticationMethod)
+            {
+                case AuthenticationMethod.Connectionstring:
+                    if (string.IsNullOrEmpty(source.ConnectionString))
+                        throw new Exception("Connection string required.");
+                    blobContainerClient = new BlobContainerClient(source.ConnectionString, source.ContainerName);
+                    break;
+
+                case AuthenticationMethod.Sastoken:
+                    if (string.IsNullOrEmpty(source.SasToken) || string.IsNullOrEmpty(source.Uri))
+                        throw new Exception("SAS Token and URI required.");
+                    authSas = true;
+                    blobContainerClient = new BlobContainerClient(new Uri($"{source.Uri}/{source.ContainerName}?"), new AzureSasCredential(source.SasToken));
+                    break;
+            }
+
+            if (source.FlatBlobListing)
+            {
+                var enumerable = blobContainerClient.GetBlobsAsync(BlobTraits.None, BlobStates.None, string.IsNullOrWhiteSpace(source.Prefix) ? null : source.Prefix).AsPages();
+                var enumerator = enumerable.GetAsyncEnumerator();
+                var flatBlobListing = await FlatBlobListing(enumerator, source, blobsList, authSas);
+                return new Result { BlobList = flatBlobListing };
+
+            }
+            else
+            {
+                var enumerable = blobContainerClient.GetBlobsByHierarchyAsync(BlobTraits.None, BlobStates.None, "/", string.IsNullOrWhiteSpace(source.Prefix) ? null : source.Prefix).AsPages();
+                var enumerator = enumerable.GetAsyncEnumerator();
+                var listBlobsHierarchy = await ListBlobsHierarchy(enumerator, source, blobsList, authSas);
+                return new Result { BlobList = listBlobsHierarchy };
+            }
+        }
+
+
+        /// <summary>
+        ///     List blobs in a flat listing structure.
+        /// </summary>
+        /// <param name="source">Source values.</param>
+        /// <param name="blobsList">List of blobs in container.</param>
+        /// <param name="authSas">Authentication method.</param>
+        /// <returns>List of blobs in container with flat listing structure.</returns>
+        public static async Task<List<Result>> FlatBlobListing(IAsyncEnumerator<Page<BlobItem>> enumerator, Source source, List<Result> blobsList, bool authSas)
+        {
+            try
+            {
+                while (await enumerator.MoveNextAsync())
```

Comment:
I think you should be able to pass the cancellation token to this method, since it is an asynchronous method.